horace_core/sqw/@sqw/change_crystal.m

```diff
@@ -1,61 +1,55 @@
 %
-% Most commonly:
-%   >> wout = change_crystal (w, rlu_corr)              % change lattice parameters and orientation
-%
-% OR
-%   >> wout = change_crystal (w, alatt)                 % change just length of lattice vectors
-%   >> wout = change_crystal (w, alatt, angdeg)         % change all lattice parameters
-%   >> wout = change_crystal (w, alatt, angdeg, rotmat) % change lattice parameters and orientation
-%   >> wout = change_crystal (w, alatt, angdeg, u, v)   % change lattice parameters and redefine u, v
-%
 %
+%   >> obj=change_crystal(obj,alignment_info)
 % Input:
 % -----
-%   w           Input sqw object
+%   w           Input sqw object or array of sqw objects
+%
+%   >> obj=change_crystal(obj,alignment_info)
+%
+% obj            -- initialized instance of Experiment object
 %
-%   rlu_corr    Matrix to convert notional rlu in the current crystal lattice to
-%              the rlu in the the new crystal lattice together with any re-orientation
-%              of the crystal. The matrix is defined by the matrix:
-%                       qhkl(i) = rlu_corr(i,j) * qhkl_0(j)
-%               This matrix can be obtained from refining the lattice and
-%              orientation with the function refine_crystal (type
-%              >> help refine_crystal  for more details).
-% *OR*
-%   alatt       New lattice parameters [a,b,c] (Angstroms)
-%   angdeg      New lattice angles [alf,bet,gam] (degrees)
-%   rotmat      Rotation matrix that relates crystal Cartesian coordinate frame of the new
-%              lattice as a rotation of the current crystal frame. Orthonormal coordinates
-%              in the two frames are related by
-%                   v_new(i)= rotmat(i,j)*v_current(j)
-%   u, v        Redefine the two vectors that were used to determine the scattering plane
-%              These are the vectors at whatever disorientation angles dpsi, gl, gs (which
-%              cannot be changed).
+% alignment_info -- helper class, containing the information
+%                   about the crystal alignment, returned by refine_crystal
+%                   routine. Type:
+%                  >> help refine_crystal  for more details.
 %
 % Output:
 % -------
-%   wout        Output sqw object with changed crystal lattice parameters and orientation
+%   wout        Output sqw object with changed crystal lattice parameters
+%               and orientation
 %
 % NOTE
-%  The input data set(s) can be reset to their original orientation by inverting the
-%  input data e.g.
-%    - call with inv(rlu_corr)
-%    - call with the original alatt, angdeg, u and v
+%  The input data set(s) can be reset to their original orientation by
+%  nullifying alignment matrix in PixelData class and setting lattice
+%  parameters to their initial values
 
 % This routine is also used to change the crystal in sqw files, when it overwrites the input file.
 
+if ~isa(alignment_info,'crystal_alignment_info') || nargin>2
```

Comment:
`nargin` is not allowed to be more than 2. There is no varargin

Reply:
now it is. And old interface have it